src/components/tasks/card/TaskStatusEditMode.tsx

```diff
@@ -1,42 +1,84 @@
 import { BACKEND_TASK_STATUS } from '@/constants/task-status';
-import task from '@/interfaces/task.type';
+import task, { CardTaskDetails } from '@/interfaces/task.type';
+import { useState } from 'react';
+import { SmallSpinner } from './SmallSpinner';
+import { SavedCheckmark } from './SavedCheckmark';
+import { ShowError } from './ShowError';
+import classNames from '@/components/tasks/card/card.module.scss';
+import { PENDING, SAVED, ERROR_STATUS } from '../constants';
+import { useUpdateTaskMutation } from '@/app/services/tasksApi';
+import { useRouter } from 'next/router';
 
 type Props = {
     task: task;
-    updateTask: (changeId: string, changeObject: { status: string }) => void;
+    setEditedTaskDetails: React.Dispatch<React.SetStateAction<CardTaskDetails>>;
 };
 
 // TODO: remove this after fixing the card beautify status
 const beautifyStatus = (status: string) => status.split('_').join(' ');
 const taskStatus = Object.entries(BACKEND_TASK_STATUS);
 
-const TaskStatusEditMode = ({ task, updateTask }: Props) => {
+const TaskStatusEditMode = ({ task, setEditedTaskDetails }: Props) => {
+    const router = useRouter();
+    const { dev } = router.query;
+
+    const [saveStatus, setSaveStatus] = useState('');
+    const [updateTask] = useUpdateTaskMutation();
+
     const onChangeUpdateTaskStatus = ({
         target: { value },
     }: React.ChangeEvent<HTMLSelectElement>) => {
-        updateTask(task.id, {
-            status: value,
+        setSaveStatus(PENDING);
+        setEditedTaskDetails((prev: any) => ({ ...prev, status: value }));
```

Comment:
any reasons specific reasons to use any ?

Reply:
Thanks for pointing it out, I have now updated it with correct type.